src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -136,6 +137,11 @@ public class ProgrammingExercise extends Exercise {
     @Column(name = "release_tests_with_example_solution", table = "programming_exercise_details")
     private boolean releaseTestsWithExampleSolution;
 
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "local_ci_build_plan_id", table = "programming_exercise_details")
+    @JsonIgnoreProperties("programmingExercises")
+    private LocalCIBuildPlan localCIBuildPlan;
```

Comment:
#6284 already introduces a `buildPlan` relation, we should not add another `localCIBuildPlan`.

Instead, the new class should be a `BuildPlan` subclass and use the common methods and interfaces. This way we don't introduce a direct coupling between the exercises and the used CI system. 

This also applies for `LocalCIBuildPlanRepository`. We should use one `BuildPlanRepository` for the regular cases.